src/core/frameworks/blocknative/Onboard.ts

```diff
@@ -139,8 +139,12 @@ export class BlocknativeWalletImpl implements Wallet {
     const { NETWORK_SETTINGS } = getConfig();
     const networkSettings = NETWORK_SETTINGS[network];
     const networkId = networkSettings.networkId;
+
     if (this.onboard) {
       this.onboard.config({ networkId });
+    }
+
+    if (this.onboard && this.name === 'MetaMask') {
```

Comment:
```suggestion
    if (this.onboard && this.name?.toUpperCase() === 'MetaMask'.toUpperCase()) {
```

Just in case :)

Reply:
Refactored in https://github.com/yearn/yearn-finance-v3/pull/559/commits/9dae939b9d53fc441c227bbf8bbdfd8d2a58dfba 👍